### PR TITLE
Only store alert hashes when iterated from localhost

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

For every child connected to a parent, the parent will read the stock and user alert configurations, and will try to create alerts for each host from them. This is maybe an optimization we could do at some point.

This PR though reduces the calculation and storing of alert hashes only when the health config iteration is for the localhost. There's no need to do it again for every child.

##### Test Plan

Start with a clean new agent. Run it, and do a select in sqlite in table `alert_hash`.
Delete the data from the table, apply this PR and re-run.
The table now will have some more hashes than before. Those are configurations that where not applied before because of `os` or `host` values. (Before this PR, if an alert configuration was not to be applied to this specific host, it would not get written in sqlite. We don't mind a few extra hashes).

### CODE REVIEW

File: health/health_config.c [hunk 1]
```diff
@@ -662,15 +662,19 @@ static int health_readfile(const char *filename, void *data) {
 
         if(hash == hash_alarm && !strcasecmp(key, HEALTH_ALARM_KEY)) {
             if(rc) {
-                if(ignore_this || !alert_hash_and_store_config(rc->config_hash_id, alert_cfg) || !rrdcalc_add_alarm_from_config(host, rc)) {
+                if (host == localhost)
```
Comment (stelfrag) [inline health/health_config.c:665]: What happens if you have health disabled on the parent but enabled for the children?

Comment (MrZammler) [reply health/health_config.c:665]: You are of course right. Will change.

**Comment by MrZammler** [comment]
I've changed the PR, and we must make sure that:

On every start or health reload of the agent, on the first iteration of the config from whatever host, the hashes will be stored. This should be evident by the field `date_updated` of the table `alert_hash` that it is updated.

On that single pass, it should store configs even for hosts or os'es that do not apply to the host in the current iteration. I.e. if the parent is linux, and the child is FreeBSD, on the parent's iteration it should also store the alert configs that are marked only for FreeBSD.

On every other pass of the configuration files, it should compute the hash, but not attempt to store it again.

**Review by thiagoftsm — APPROVED** [review]
I ran tests with this PR and for this I used a parent and a child with compiled with this branch.

I observed that timestamp was filled when I start the parent and when I ran `netdatacli health-reload`. When child connected to host, the table `host` was updated, but nothing was changed in `alert_hash`.

LGTM!

**Review by stelfrag — APPROVED** [review]

**Comment by MrZammler** [comment]
Thanks!

**Pull request merged**